[chip-test, sival] ENTROPY_SRC SiVal test plan

vogelpi opened the pull request:
This is work in progress. I hope to continue this on THU. FYI @marnovandermaas .

What's missing:
- Finish description of features.
- Check what existing TLTs cover already / how they can be modified to cover what's missing.
- Check if new tests are needed.

marnovandermaas commented:
I added a test `chip_sw_entropy_src_fips_health_check` to cover more of the features, but that last commit is still very much a work in progress.

Yet to be covered features:
- `ENTROPY_SRC.RNG_BIT_ENABLE`
- `ENTROPY_SRC.MODE.BYPASS`
- `ENTROPY_SRC.HEALTH_TESTS_EXT`

vogelpi commented:
Thanks for your updates @marnovandermaas . I now have some time to look further into this and will force-push again once I am done.

vogelpi commented:
CHANGE AUTHORIZED: hw/ip/entropy_src/data/entropy_src.hjson

vogelpi commented:
I've now completed the feature list. Besides describing the main modes of operation (FIPS mode, bypass mode) and the firmware override modes in more detail, I've converted the `ROUTE_TO_FIRMWARE` feature to a non-mode feature and removed the CSRNG mode you've added @marnovandermaas . The reason is that these two settings are actually orthogonal to the FIPS and bypass modes and I thought they shouldn't be mixed up.

I've also reviewed some of the existing tests and removed an outdated TODO + fixed some test descriptions. People seem to have confused the the firmware override modes, the route-to-firmware feature, the boot-time / bypass mode and the hardware conditioning bypass feature. I've aligned the descriptions with actual test implementations and updated the linked features.

When doing that, I also noted that the existing tests only cover quite little of the core features of `entropy_src`. I've thus added two more tests and extended the test you drafted @marnovandermaas . Now, also `ENTROPY_SRC.RNG_BIT_ENABLE` and `ENTROPY_SRC.MODE.BYPASS` are covered.

`ENTROPY_SRC.HEALTH_TESTS_EXT` isn't covered as Earl Grey does not support this (see feature description).

For the test requiring fuses, I assumed that we need to run them in TEST_UNLOCKED although PROD might be sufficient (see our [internal discussion regarding the corresponding OTP fuses](https://github.com/lowRISC/opentitan-integrated/pull/319/files#r1303295142)).

This is now ready for review.

vogelpi commented on hw/top_earlgrey/data/ip/chip_entropy_src_testplan.hjson:
```diff
@@ -0,0 +1,195 @@
+            - Enable ENTROPY_SRC in FIPS mode.
+            - Enable Firmware Override - Extract & Insert to route entropy received from AST to the observe FIFO.
+            - Verify that the observe FIFO depth is non-zero via SW - indicating the reception of data over the AST RNG interface.
+            - Verify the correctness of the received data with assertion based connectivity checks.
```
We can't check whether the values match in silicon. But checking that the Observe FIFO fills up is a very fundamental test prior to validation testing. Since the test is already there I've marked this as SV1. This is assuming Nuvoton will use this for validating the noise source.

andreaskurth replied:
I think it's worth mentioning this in the *Notes for silicon targets* below

vogelpi replied:
Good idea. I've added this information there as well.

vogelpi commented on hw/top_earlgrey/data/ip/chip_entropy_src_testplan.hjson:
```diff
@@ -0,0 +1,195 @@
+        "ENTROPY_SRC.RNG_BIT_ENABLE",
+      ]
+      stage: V3
+      si_stage: SV1
```
I've created this test to provide a skeleton usable by Nuvoton for validating the noise source's channels.

vogelpi commented on hw/top_earlgrey/data/ip/chip_entropy_src_testplan.hjson:
```diff
@@ -0,0 +1,195 @@
+            - Verify that the observe FIFO depth is non-zero and read out post-health test entropy for validation testing.
+            - Repeat the procedure four times with the ENTROPY_SRC.RNG_BIT_ENABLE feature being enabled (once for every channel).
+
+            Notes for silicon targets:
+            - This test requires access to the ENTROPY_SRC.FW_OV.OBSERVE feature, which is gated by OTP.
+              This test might thus need to be done in a secure facility.
```
Like other tests, we might also be able to run this in PROD eventually. Depending on when we're going to activate the corresponding OTP fuses exactly.

andreaskurth commented on hw/top_earlgrey/data/ip/chip_entropy_src_testplan.hjson:
```diff
@@ -61,9 +74,122 @@
+            - Verify that all entropy consuming endpoints can finish their operations and do not hang.
+            - Repeat the procedure with different health test window sizes for FIPS mode.
+            - Disable the entropy complex.
+            - Configure very stringent health test threshold values for FIPS mode.
```
Suggestion: replace "very" by "unrealistically" to align with wording above

vogelpi replied:
Done, thanks.

andreaskurth reviewed (approved):
LGTM with two minor suggestions; thanks @vogelpi!

vogelpi commented:
Thanks @andreaskurth for your review. I've implemented your suggestions and rebased this PR once more on master.

andreaskurth commented:
CHANGE AUTHORIZED: hw/ip/entropy_src/data/entropy_src.hjson
